src/theory/theory_preprocessor.cpp

```diff
@@ -122,6 +121,23 @@ TrustNode TheoryPreprocessor::preprocessInternal(
     }
     Trace("tpp-debug") << "TheoryPreprocessor::preprocess: finish" << std::endl;
   }
+
+  if (procLemmas)
+  {
+    // Also must preprocess the new lemmas. This is especially important for
+    // formulas containing witness terms whose bodies are not in preprocessed
+    // form, as term formula removal introduces new lemmas for these that
+    // require theory-preprocessing.
```

Comment:
Is this the reason why we can't simply push the lemmas to the inference manager?

Reply:
Yes, because this is being done at a time when the inference manager is not ready.  In particular, ppRewrite is called during preprocessing where the SAT solver has not yet received any assertions.